### ISSUE TITLE
Make elyra installation work on PIP >= 20.3

### ISSUE THREAD

**Issue by romeokienzler** [elyra-ai/elyra#2648]
**Is your feature request related to a problem? Please describe.**
RedHat can't build elyra images with legacy-resolver or PIP < 20.3 

Please see [documentation](https://elyra.readthedocs.io/en/latest/getting_started/installation.html#pip)

**Describe the solution you'd like**
Make elyra installation work on PIP >= 20.3

**Describe alternatives you've considered**
Convince RedHat to change their build to cope with this limitation

**Comment by ptitzler** [comment]
Can you provide the errors they are getting?

**Comment by guimou** [comment]
Hi!
This is what we get when installing Elyra in our base image:
```
ERROR: Cannot install -r requirements.txt (line 214), -r requirements.txt (line 81) and click==8.0.4 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested click==8.0.4
    black 22.1.0 depends on click>=8.0.0
    elyra 3.5.1 depends on click<8 and >=7.1.1

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```
Pip is at version 22.0.4 in this base image.
The whole requirements.txt is automatically generated with hashes after checks by out internal "advisor" tool. 

Normally this is a know error, at least since 2020, as per Elyra documentation: https://elyra.readthedocs.io/en/latest/getting_started/installation.html#pip

And indeed the workaround with `--use-deprecated legacy-resolver` works.
But we won't revert to an old version of pip nor introduce this workaround which has other impact on the whole build chain only for this package.

Adding [requirements.txt](https://github.com/elyra-ai/elyra/files/8445584/requirements.txt)

**Comment by kevin-bates** [comment]
FWIW, this particular cap has been removed in [Elyra 3.7](https://github.com/elyra-ai/elyra/releases/tag/v3.7.0) if that makes things easier.

**Comment by guimou** [comment]
Lol, it was pushed last week! You were just waiting for us to open the issue, no?

**Comment by guimou** [comment]
Sorry if it was misinterpreted (from your confused Emoji). I just had a good (happy) laugh because it's the second time this year that something that had been bothering me for months ends up beind corrected while I was not paying attention.
So definitely no complaint or anything, totally the opposite. I was pretty sure of myself we had this issue when bringing this up in a meeting this morning because I had worked on this just before the fresh new release. And as the warning was still there in the docs I did not check again this week.
Sorry for the confusion, there was again no sarcasm intented in my comment.

**Comment by kevin-bates** [comment]
Sorry about that, I didn't realize I hit the frown/confused emoji (the font was too small).  I've updated to the originally intended smiley face and apologize that you thought I thought otherwise.  It's all good. :smile: :+1:

**Comment by akchinSTC** [comment]
@guimou - Just to clarify, is this issue related to opendatahub at all? or is the strictly have to do with RHEL images?

**Comment by romeokienzler** [comment]
@kevin-bates this problem occurs independently from opendatahub, when building ubi based images for e.g. opendatahub

**Comment by romeokienzler** [comment]
@guimou correct?

**Comment by guimou** [comment]
Yes, this happens whenever you have a recent pip version. So not only ubi8, but also Fedora,... I think it should be the same for any OS if you use pip 20.3 or later as that's what Elyra documentation says. But I have not tested for other setup.
That said, if it's corrected in 3.7.0, maybe the only thing to do is to update the documentation.

**Comment by ptitzler** [comment]
The only change that's included in version 3.7 is that some package version caps were lifted. The legacy resolver still needs to be used.

**Comment by guimou** [comment]
Ah, sorry, I did not get it right. Then the issue is still there I'm afraid.

**Comment by ptitzler** [comment]
So you did try to install using Elyra version 3.7 and are still getting the unresolvable conflicts?

**Comment by guimou** [comment]
Ok, I can confirme it works without the old legacy resolver in my base image.
So I guess the doc can be amended to indicate that the problem has been resolved since 3.7.0.
Thanks a lot!

**Comment by romeokienzler** [comment]
documentation updated in #2654
